server/master/src/main/java/org/apache/accumulo/master/tserverOps/ShutdownTServer.java

```diff
@@ -68,8 +63,6 @@ public long isReady(long tid, Master master) throws Exception {
           TabletServerStatus status = connection.getTableMap(false);
           if (status.tableMap != null && status.tableMap.isEmpty()) {
             log.info("tablet server hosts no tablets " + server);
-            connection.halt(master.getMasterLock());
```

Comment:
This may have been called in isReady intentionally in-order to get some re-try type behavior, not sure.  The command later on in the method made me think this.

Reply:
Alternatively, in addition to sending a halt we could remove the zookeeper lock here.

Reply:
Working through the logic, I believe the halt was actually better placed in the isReady call.  Reverting that part of the change.